src/runtime_src/hip/core/module.cpp

```diff
@@ -33,6 +44,18 @@ module_elf(module_xclbin* xclbin_module, const std::string& file_name)
 {
 }
 
+module_elf::
+module_elf(module_xclbin* xclbin_module, void* data, size_t size)
+  : module{xclbin_module->get_context(), false}
+  , m_xclbin_module{xclbin_module}
+{
+  std::istringstream stream;
+  stream.rdbuf()->pubsetbuf(static_cast<char*>(data), size);
+
+  m_xrt_elf = xrt::elf{stream};
+  m_xrt_module = xrt::module{m_xrt_elf};
```

Comment:
ditto, no ctor body, use static helper functions as necessary. Why?  Because the stream created here is for the purpose of ctoring the elf only, it has nothing to do with m_xrt_module.